### ISSUE TITLE
Importing node pools can forces replacement

### ISSUE THREAD

**Issue by caueasantos** [terraform-google-modules/terraform-google-kubernetes-engine#2004]
### TL;DR

The module is adding two key-value pairs `cluster_name` and `node_pool` that triggers a replacement if the node pool was not provisioned using the community module.

GKE [node_pools_metadata](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/autogen/main/cluster.tf.tmpl#L894)

https://github.com/hashicorp/terraform-provider-google/issues/1388

### Expected behavior

Allow users to fully overwrite node pools metadata or expose a way to disable key-pair values added by default

### Observed behavior

The node pool is marked to be replaced
```terraform
            ...
            local_ssd_count             = 0
            logging_variant             = "DEFAULT"
            machine_type                = "t2d-standard-16"
          ~ metadata                    = { # forces replacement
              + "cluster_name"             = "admin"
                "disable-legacy-endpoints" = "true"
              + "node_pool"                = "runners"
            ...
            }
```

### Terraform Configuration

```hcl
import {
  id = "projects/project-id/locations/us-east1/clusters/admin"
  to = module.cluster.module.gke.google_container_cluster.primary
}

import {
  id = "project-id/us-east1/admin/runners"
  to = module.cluster.module.gke.google_container_node_pool.pools["runners"]
}

module "cluster" {
  source                     = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"
  version                    = "30.0.0"
  ip_range_pods              = var.ip_range_pods
  ip_range_services          = var.ip_range_services
  master_ipv4_cidr_block     = var.master_ipv4_cidr_block
  name                       = "admin"
  network                    = "default"
  project_id                 = "project-id"
  region                     = "us-east1"
  subnetwork                 = "default"
  deletion_protection        = true
  master_authorized_networks = var.master_authorized_networks

  node_pools = [
    {
      name               = "runners"
      machine_type       = "t2d-standard-16"
      initial_node_count = 0
      min_count          = 0
      max_count          = 20
      enable_secure_boot = true
      spot               = true
      location_policy    = "ANY"
      effect             = "NO_SCHEDULE"
      key                = "node_type"
      value              = "runners"
    }
  ]
}
```

### Terraform Version

```sh
Terraform v1.7.4 on darwin_arm64
```
